sid/tests/test_contacts.py

```diff
@@ -184,3 +186,144 @@ def test_calculate_infections():
+
+@pytest.fixture
+def contact_models():
+    def meet_one(states, params):
+        return pd.Series(1, index=states.index)
+
+    def first_half_meet(states, params):
+        n_contacts = pd.Series(0, index=states.index)
+        first_half = round(len(states) / 2)
+        n_contacts[:first_half] = 1
+        return n_contacts
+
+    contact_models = {
+        "meet_one": {
+            "model": meet_one,
+            "is_recurrent": False,
+        },
+        "first_half_meet": {
+            "model": first_half_meet,
+            "is_recurrent": False,
+        },
+    }
+    return contact_models
+
+
+def test_calculate_contacts_no_policy(states_all_alive, contact_models):
+    contact_policies = {}
+    date = pd.Timestamp("2020-09-29")
+    params = pd.DataFrame()
+    first_half = round(len(states_all_alive) / 2)
+    expected = np.array(
+        [[1, i < first_half] for i in range(len(states_all_alive))],
+        dtype=DTYPE_N_CONTACTS,
+    )
+    res = calculate_contacts(
+        contact_models=contact_models,
+        contact_policies=contact_policies,
+        states=states_all_alive,
+        params=params,
+        date=date,
+    )
+    np.testing.assert_array_equal(expected, res)
+
+
+def test_calculate_contacts_policy_inactive(states_all_alive, contact_models):
+    contact_policies = {
+        "first_half_meet": {
+            "start": "2020-08-01",
+            "end": "2020-08-30",
+            "is_active": lambda states: True,
+            "multiplier": 0.0,
+        },
+    }
+    date = pd.Timestamp("2020-09-29")
+    params = pd.DataFrame()
+    first_half = round(len(states_all_alive) / 2)
+    expected = np.array(
+        [[1, i < first_half] for i in range(len(states_all_alive))],
+        dtype=DTYPE_N_CONTACTS,
+    )
+    res = calculate_contacts(
+        contact_models=contact_models,
+        contact_policies=contact_policies,
+        states=states_all_alive,
+        params=params,
+        date=date,
+    )
+    np.testing.assert_array_equal(expected, res)
+
+
+def test_calculate_contacts_policy_active(states_all_alive, contact_models):
+    contact_policies = {
+        "first_half_meet": {
+            "start": "2020-09-01",
+            "end": "2020-09-30",
+            "is_active": lambda states: True,
+            "multiplier": 0.0,
+        },
+    }
+    date = pd.Timestamp("2020-09-29")
+    params = pd.DataFrame()
+    expected = np.array(
+        [[1, 0] for _ in range(len(states_all_alive))],
+        dtype=DTYPE_N_CONTACTS,
+    )
+    res = calculate_contacts(
+        contact_models=contact_models,
+        contact_policies=contact_policies,
+        states=states_all_alive,
+        params=params,
+        date=date,
+    )
+    np.testing.assert_array_equal(expected, res)
+
+
+@pytest.fixture
+def states_with_dead(states_all_alive):
+    states_all_alive.loc[:2, "dead"] = [True, False, True]
+    states_all_alive.loc[5:, "needs_icu"] = [True, False, True]
+    return states_all_alive
```

Comment:
Potential problem with inplace errors if the fixture will not be recreated every function call. Just to be on the safe side for the future.
```suggestion
    states_with_dead = states_all_alive.copy()
    states_with_dead.loc[:2, "dead"] = [True, False, True]
    states_with_dead.loc[5:, "needs_icu"] = [True, False, True]
    return states_with_dead
```